packages/bpk-component-price/src/BpkPrice.js

```diff
@@ -54,35 +63,60 @@ const BpkPrice = (props: Props) => {
       // $FlowFixMe[cannot-spread-inexact] - inexact rest. See 'decisions/flowfixme.md'.
       {...rest}
     >
-      {subtitle && (
-        <BpkText
-          className={getClassName('bpk-price__subtitle')}
-          textStyle={isSmall ? TEXT_STYLES.xs : TEXT_STYLES.sm}
-          tagName="span"
-        >
-          {subtitle}
-        </BpkText>
-      )}
+      <div
+        className={getClassName(
+          previousPrice && 'bpk-price__leadingText',
+          isAlignRight && 'bpk-price__leadingText--right',
+        )}
+      >
+        {previousPrice && (
+          <BpkText
+            className={getClassName('bpk-price__previous-price')}
+            textStyle={isSmall ? TEXT_STYLES.xs : TEXT_STYLES.sm}
+            tagName="span"
+          >
+            {previousPrice}
+          </BpkText>
+        )}
+        {previousPrice && (
+          <BpkText
+            textStyle={isSmall ? TEXT_STYLES.xs : TEXT_STYLES.sm}
+            tagName="span"
+            className={getClassName('bpk-price__separator')}
+          >
+            &#67871;
+          </BpkText>
+        )}
+
+        {leadingText && (
+          <BpkText
+            textStyle={isSmall ? TEXT_STYLES.xs : TEXT_STYLES.sm}
+            tagName="span"
+          >
+            {leadingText}
+          </BpkText>
```

Comment:
Probably not worth it because of it creates an extra layer that will just wrap our Text component in the same way. So would then increase overhead and maintenance/complexity to the system. So to me doesn't make sense to just move this same logic to somewhere else.

Which then we would have to do massive change across our components to follow this pattern as if we introduce it once we have to introduce it everywhere

Reply:
Agree!